### PR TITLE
Internal: Make AI promotions less intrusive [ED-14582]

### PR DESCRIPTION
**Pull request by DennisNerush**

### CODE REVIEW

File: modules/ai/assets/js/editor/ai-behavior.js [hunk 1]
```diff
@@ -146,7 +146,7 @@ export default class AiBehavior extends Marionette.Behavior {
 			return;
 		}
 
-		if ( ! shouldShowPromotionIntroduction() ) {
+		if ( ! shouldShowPromotionIntroduction( sessionStorage ) ) {
```
Comment (netanelavr) [inline modules/ai/assets/js/editor/ai-behavior.js:149]: where this `sessionStorage` is coming from?

File: modules/ai/assets/js/editor/utils/promotion-introduction-session-validator.js [hunk 2]
```diff
@@ -1,8 +1,22 @@
...
-	const editorSessionValue = sessionStorage.getItem( 'ai_promotion_introduction_editor_session_key' );
-	if ( ! editorSessionValue || editorSessionValue !== EDITOR_SESSION_ID ) {
-		sessionStorage.setItem( 'ai_promotion_introduction_editor_session_key', EDITOR_SESSION_ID );
+function passedRequiredHoursSinceLastPromotion( currentTimestamp, lastPromotionTimestamp ) {
+	return currentTimestamp - parseInt( lastPromotionTimestamp ) >= HOURS_DELTA_IN_MILLISECONDS;
+}
+
+export function shouldShowPromotionIntroduction( session ) {
+	const editorSessionValue = session.getItem( 'ai_promotion_introduction_editor_session_key' );
+	const lastPromotionTimestamp = editorSessionValue?.split( '#' )[ 1 ];
```
Comment (netanelavr) [inline modules/ai/assets/js/editor/utils/promotion-introduction-session-validator.js:10]: is it possible a session id will contain a hash mark or its a uuid style?

**Review by netanelavr — DISMISSED** [review]

File: modules/ai/assets/js/editor/ai-behavior.js (hunk 1 above)
Comment (DennisNerush) [reply modules/ai/assets/js/editor/ai-behavior.js:149]: window context

File: tests/jest/unit/modules/ai/assets/js/editor/utils/promotion-introduction-session-validator.test.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+		window.EDITOR_SESSION_ID = 'editor-session-123';
+		mockSession.getItem.mockReturnValue( 'editor-session-123' );
+
+		const result = shouldShowPromotionIntroduction( mockSession );
+
+		expect( result ).toBe( false );
+	} );
+
+	it( `Should return false when previous promotion appeared less than ${ HOURS_BETWEEN_PROMOTION_INTRODUCTIONS } hours ago`, async () => {
+		const currentDate = new Date( 2024, 5, 3, 15 );
```
Comment (matipojo) [inline tests/jest/unit/modules/ai/assets/js/editor/utils/promotion-introduction-session-validator.test.js:32]: The test will not work after Jun, 03 2024

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — DISMISSED** [review]

**Review by netanelavr — APPROVED** [review]

**Pull request merged**